### PR TITLE
Lots of cppcheck cleanups still in progress

### PR DESCRIPTION
**Pull request by mdblack98**
I tried to be conservative on #ifdef'ing unreferenced functions some are obviously referenced from non-hamlib applications like callback functions.
Hopefully didn't remove something somebody needed.
Also a few bug fixes in here
And a memory leak in gpio.c fixed.

### CODE REVIEW

File: kenwood/elecraft.c [hunk 1]
```diff
@@ -96,11 +96,6 @@ int elecraft_open(RIG *rig)
     rig_debug(RIG_DEBUG_VERBOSE, "%s called, rig version=%s\n", __func__,
               rig->caps->version);
 
-    if (!rig)
-    {
-        return -RIG_EINVAL;
-    }
-
```
Comment (N0NB) [inline kenwood/elecraft.c]: I know cppchecker complained about these tests.  They've been in there as long as I can recall and maybe they offer no protection and are simply redundant.  I suppose my ignorance/forgetfulness is showing!

File: src/locator.c [hunk 2]
```diff
@@ -426,8 +425,8 @@ int HAMLIB_API locator2longlat(double *longitude,
     /* For x(=longitude) and y(=latitude) */
     for (x_or_y = 0;  x_or_y < 2;  ++x_or_y)
     {
-        ordinate = -90.0;
-        divisions = 1;
+        double ordinate = -90.0;
+        int divisions = 1;
```
Comment (N0NB) [inline src/locator.c]: Just a question, will Bill raise an issue that moving these declarations down into the code block will cause problems with some C89 compilers as he did a while back?

**Comment by mdblack98** [comment]
Oops...as I was doing this I was thinking it was just declarations in for loops.
Bill did test all 20 compilers he was using but not sure if it included mixed declarations on his last test.
I'll fix these....



#include <stdio.h>int main(){  printf("Testing\n");  int i;  for(i=0;i<5;++i) printf("i=%d\n",i);  for(int j=0;j<5;++j) printf("j=%d\n",j);  return 0;}

root@osboxes:/home/osboxes/mike/hamlib/kenwood# cc -Wall -pedantic -o x x.croot@osboxes:/home/osboxes/mike/hamlib/kenwood# cc -std=c99 -Wall -pedantic -o x x.croot@osboxes:/home/osboxes/mike/hamlib/kenwood# cc -std=c89 -Wall -pedantic -o x x.cx.c: In function ‘main’:x.c:5:3: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]   int i;   ^~~x.c:7:3: error: ‘for’ loop initial declarations are only allowed in C99 or C11 mode   for(int j=0;j<5;++j) printf("j=%d\n",j);   ^~~x.c:7:3: note: use option -std=c99, -std=gnu99, -std=c11 or -std=gnu11 to compile your code




 

    On Monday, December 9, 2019, 11:08:56 AM CST, Nate Bargmann <notifications@github.com> wrote:  
 
 
@N0NB commented on this pull request.

In src/locator.c:
> @@ -426,8 +425,8 @@ int HAMLIB_API locator2longlat(double *longitude,
     /* For x(=longitude) and y(=latitude) */
     for (x_or_y = 0;  x_or_y < 2;  ++x_or_y)
     {
-        ordinate = -90.0;
-        divisions = 1;
+        double ordinate = -90.0;
+        int divisions = 1;
 

Just a question, will Bill raise an issue that moving these declarations down into the code block will cause problems with some C89 compilers as he did a while back?

**Comment by N0NB** [comment]
Might have Bill check this over before undoing everything you're done.

**Comment by mdblack98** [comment]
Not a problem....I think it's worthwhile to make it C90 compatible...even if that is 29 years ago :-)
Most of the changes are not ones I made actually.  Over 500 warnings that have been in hamlib for quite a while.
Mike

 

    On Monday, December 9, 2019, 12:29:43 PM CST, Nate Bargmann <notifications@github.com> wrote:  
 
 
Might have Bill check this over before undoing everything you're done.

**Comment by mdblack98** [comment]
I'm going to hold off on changes that might impact Malcom's Icom patch he submitted.
Can you get his integrated into the master?

 

    On Monday, December 9, 2019, 11:08:56 AM CST, Nate Bargmann <notifications@github.com> wrote:  
 
 
@N0NB commented on this pull request.

In src/locator.c:
> @@ -426,8 +425,8 @@ int HAMLIB_API locator2longlat(double *longitude,
     /* For x(=longitude) and y(=latitude) */
     for (x_or_y = 0;  x_or_y < 2;  ++x_or_y)
     {
-        ordinate = -90.0;
-        divisions = 1;
+        double ordinate = -90.0;
+        int divisions = 1;
 

Just a question, will Bill raise an issue that moving these declarations down into the code block will cause problems with some C89 compilers as he did a while back?

**Comment by N0NB** [comment]
I have yours pulled into a local branch so I can merge his request now.

**Comment by N0NB** [comment]
* On 2019 09 Dec 12:33 -0600, Michael Black wrote:
> Not a problem....I think it's worthwhile to make it C90
> compatible...even if that is 29 years ago :-)

I was thinking we could shoot for C99 compliance earlier in the year
until Bill commented on certain compilers not being C99 capable.

> Most of the changes are not ones I made actually.  Over 500 warnings
> that have been in hamlib for quite a while.

I understand.  I applaud you for diving into that mess!

73, Nate

-- 

"The optimist proclaims that we live in the best of all
possible worlds.  The pessimist fears this is true."

Web: https://www.n0nb.us
Projects: https://github.com/N0NB
GPG fingerprint: 82D6 4F6B 0E67 CD41 F689 BBA6 FB2C 5130 D55A 8819

**Comment by N0NB** [comment]
I see this on my MinGW 32 and 64 bit builds:

```
  CC       parallel.lo
parallel.c: In function ‘par_open’:
parallel.c:130:9: warning: unused variable ‘mode’ [-Wunused-variable]
     int mode;
         ^~~~
```

Native builds on Debian 10 with GCC and clang are clean.

**Comment by mdblack98** [comment]
Fixed...
de Mike W9MDB

 

    On Monday, December 30, 2019, 08:04:59 PM CST, Nate Bargmann <notifications@github.com> wrote:  
 
 
I see this on my MinGW 32 and 64 bit builds:
  CC       parallel.lo
parallel.c: In function ‘par_open’:
parallel.c:130:9: warning: unused variable ‘mode’ [-Wunused-variable]
     int mode;
         ^~~~

Native builds on Debian 10 with GCC and clang are clean.

**Comment by N0NB** [comment]
I see this now with MinGW:

```
  CC       rigctl-rigctl.o
rigctl.c:125:18: warning: ‘have_rl’ defined but not used [-Wunused-const-variable=]
 static const int have_rl = 0;
                  ^~~~~~~
```

**Comment by mdblack98** [comment]
Fixed it -- just removed that variable entirely as it was redundant.
de Mike W9MDB

 

    On Wednesday, January 1, 2020, 06:53:11 PM CST, Nate Bargmann <notifications@github.com> wrote:  
 
 
I see this now with MinGW:
  CC       rigctl-rigctl.o
rigctl.c:125:18: warning: ‘have_rl’ defined but not used [-Wunused-const-variable=]
 static const int have_rl = 0;
                  ^~~~~~~

**Comment by N0NB** [comment]
Thanks, Mike.

I plan to get this merged into master this evening to give your changes
wider exposure.

73, Nate

-- 

"The optimist proclaims that we live in the best of all
possible worlds.  The pessimist fears this is true."

Web: https://www.n0nb.us
Projects: https://github.com/N0NB
GPG fingerprint: 82D6 4F6B 0E67 CD41 F689 BBA6 FB2C 5130 D55A 8819

**Pull request merged**